### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by salvino080-coder** [pytorch/xla#9739]
"""
Utility helpers: padding/bucketing for fixed-shape batches.

Functions:
- pad_to_fixed_shape(tensor, target_shape, pad_value=0)
- collate_fixed_shape(batch, target_shape, pad_value=0)

target_shape should be the per-sample shape excluding batch dim, e.g. (3, 224, 224).
"""
import torch

def pad_to_fixed_shape(tensor, target_shape, pad_value=0):
    """
    Pad a single tensor sample (C, H, W or H, W or similar) to target_shape.
    If tensor has fewer dims than target_shape, will raise.
    """
    if tensor.dim() < 1:
        raise ValueError("tensor must have at least 1 dim (channels/spatial).")
    src_shape = tuple(tensor.shape)
    if src_shape == target_shape:
        return tensor
    # create output tensor
    out = torch.full(target_shape, pad_value, dtype=tensor.dtype, device=tensor.device)
    # determine slices
    slices = tuple(slice(0, min(s, t)) for s, t in zip(src_shape, target_shape))
    out[slices] = tensor[slices]
    return out

def collate_fixed_shape(batch, target_shape, pad_value=0):
    """
    Collate a list of (x, y) where x is a tensor sample, to a batch with fixed shape.
    target_shape excludes batch dim, e.g. (3,224,224).
    Returns (batch_x, batch_y).
    """
    xs, ys = zip(*batch)
    padded = []
    for x in xs:
        if tuple(x.shape) != tuple(target_shape):
            padded_x = pad_to_fixed_shape(x, target_shape, pad_value)
        else:
            padded_x = x
        padded.append(padded_x)
    batch_x = torch.stack(padded, dim=0)
    batch_y = torch.stack([torch.as_tensor(y) for y in ys], dim=0)
    return batch_x, batch_y
